packages/neo-one-monitor/src/BrowserMetricsFactory.js

```diff
@@ -0,0 +1,123 @@
+/* @flow */
+import type { MetricConstruct, MetricsFactory, RawLabels } from './MonitorBase';
+
+import type { Counter, Gauge, Histogram, Summary } from './types';
+
+const metrics = {
+  counters: [],
+  histograms: [],
+};
+
+export const resetMetrics = (): void => {
+  metrics.counters = [];
+  metrics.histograms = [];
+};
+
+type MetricValue = {|
+  countOrLabels?: number | RawLabels,
+  count?: number,
+|};
+
+export class BaseMetric {
+  metric: MetricConstruct;
+  values: Array<MetricValue>;
+
+  constructor(metric: MetricConstruct) {
+    this.metric = metric;
+    this.values = [];
+  }
+
+  reset(): void {
+    this.values = [];
+  }
+}
+
+export type MetricCollection = {|
+  counters: Array<BaseMetric>,
+  histograms: Array<BaseMetric>,
+|};
+
+class BrowserCounter extends BaseMetric implements Counter {
+  inc(countOrLabels?: number | RawLabels, count?: number): void {
+    this.values.push({ countOrLabels, count });
+  }
+}
+
+export class BrowserGauge extends BaseMetric implements Gauge {
+  // eslint-disable-next-line
+  inc(countOrLabels?: number | RawLabels, count?: number): void {}
+
+  // eslint-disable-next-line
+  dec(countOrLabels?: number | RawLabels, count?: number): void {}
+
+  // eslint-disable-next-line
+  set(countOrLabels?: number | RawLabels, count?: number): void {}
+}
+
+class BrowserHistogram extends BaseMetric implements Histogram {
+  observe(countOrLabels?: number | RawLabels, count?: number): void {
+    this.values.push({ countOrLabels, count });
+  }
+}
+
+export class BrowserSummary extends BaseMetric implements Summary {
+  // eslint-disable-next-line
+  observe(countOrLabels?: number | RawLabels, count?: number): void {}
+}
+
+interface MetricsFactoryCollect extends MetricsFactory {
+  collect(): MetricCollection;
+  _serializeJSON(inputJSON: Object): Object;
+}
+
+export default class BrowserMetricsFactory implements MetricsFactoryCollect {
+  collect(): MetricCollection {
+    const currentMetrics = this._serializeJSON(metrics);
+    for (const metricType of Object.keys(metrics)) {
+      metrics[metricType].map(metric => metric.reset());
+    }
+    return currentMetrics;
+  }
+
+  createGauge(options: MetricConstruct): Gauge {
+    return new BrowserGauge(options);
+  }
+
+  createCounter(options: MetricConstruct): Counter {
+    const counter = new BrowserCounter(options);
+    metrics.counters.push(counter);
+
+    return counter;
+  }
+
+  createHistogram(options: MetricConstruct): Histogram {
+    const histogram = new BrowserHistogram(options);
+    metrics.histograms.push(histogram);
+
+    return histogram;
+  }
+
+  createSummary(options: MetricConstruct): Summary {
+    return new BrowserSummary(options);
+  }
+
+  _serializeJSON(inputJSON: Object): Object {
```

Comment:
This isn't quite what I meant by `_serializeJSON`. You should have an explicit type in and an explicit type out - not trying to implement some kind of generic jsonify function. Do something like
```
_serializeJSON(metrics: Metrics): MetricCollection
```

This also is totally broken - you're returning `MetricCollection` from `collect` when it's not matching the type at all.